zlch

zlch opened the pull request:
## Gumball

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Gumball frees your mind of having to remember things you need to do ~~stopping you from being too lazy~~. It's,

-text-based

-easy to learn

-FAST to use

All you need to do is,

1.download it from _here_.

2.double-click it.

3.add your tasks.

4.let it manage your tasks for you 😸:

Plus its completely __Free__
Features:

- [x] Managing tasks

- [x] Managing deadlines

- [ ] Reminders (coming soon)

Here is a look at the main method:

```
public static void main(String[] args) {
        try {
            Gumball chat = new Gumball();
            chat.start();
        } catch (IOException e) {
            UI.print(e.getMessage());
        } catch (InputErrorException e) {
            UI.print("Error in file please start over.");
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

zlch committed:
Week 2 level 0: Change of name, implementation of basic skeleton.

zlch committed:
Added input scanner, added echo, changed the way lines are printed, added start function.

zlch committed:
Created class for list, added ability to add and get list, added method to execute different commands based on input.

zlch committed:
Added task class, change list to specificly list for tasks, added ability to mark as done.

zlch committed:
Added different task types.

zlch committed:
Fixed small error with additional spacing in description for Deadlines and Events added Automated Text UI Testing

zlch committed:
Added some exception handling including new Exception types for Exception thrown due to incorrect input as well as Excpetion resulting specificly from Tasks.

zlch committed:
Added delete function, changed List in TaskList to an ArrayList

zlch committed:
Add ability to save the list after chat bot is closed. Add a fileManager class to deal with the file.

zlch committed:
Add dates and time recognition for deadline

zlch committed:
Merge branch 'Branch-Level-8'

zlch committed:
Change to a more OOP design

zlch committed:
Organise everything into a package

zlch committed:
Merge branch 'add-gradle-support'

zlch committed:
Add gradle functionality
Fix delete function

zlch committed:
Add JUnit tests for TaskList Task and ToDo

zlch committed:
Add javaDoc information to most functions

zlch committed:
Change code to meet coding standard

zlch committed:
Add a find command which list all tasks containing keyword

zlch committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

zlch committed:
Merge branch 'branch-A-JavaDoc'

zlch committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Gumball/Deadline.java
#	src/main/java/Gumball/Event.java
#	src/main/java/Gumball/ListCommand.java
#	src/main/java/Gumball/Task.java
#	src/main/java/Gumball/TaskList.java

zlch committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Gumball/Parser.java

sarahteonin commented on src/main/java/Gumball/AddCommand.java:
```diff
@@ -0,0 +1,51 @@
+package Gumball;
```
Should this package be in upper case?

sarahteonin commented on src/main/java/Gumball/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+            LocalDate d1 = LocalDate.parse(section[1]);
+            deadlineTime = d1;
+            super.description = section[0] + "(by: "
+                    + d1.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
I like how you remembered to include 8 space indentation for wrapped lines! 👍

sarahteonin commented on src/main/java/Gumball/Event.java:
```diff
@@ -0,0 +1,25 @@
+            super.description = section[0] + "(from: " + section[1] + " to: " + section[2] + ")";
+            taskType = "[E]";
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new TaskException("Sorry, the desctiption you " +
```
Not coding standard but is this supposed to be description?

sarahteonin commented on src/main/java/Gumball/FileManager.java:
```diff
@@ -0,0 +1,74 @@
+            } else if (command.startsWith("event")) {
+                taskList.add(new Event(command));
+            }
+            if (input.startsWith("[X]")) {
```
Should there be a white space before this if statement to enhance readibility?

sarahteonin commented on src/main/java/Gumball/FindCommand.java:
```diff
@@ -0,0 +1,14 @@
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+    @Override
```
Should there be a white space between methods?

sarahteonin commented on src/main/java/Gumball/Gumball.java:
```diff
@@ -0,0 +1,61 @@
+
+    /**
+     * Will cause the chatbot to start running.
+     * @throws FileNotFoundException
```
Should there be a empty line between description and throws?

sarahteonin commented on src/main/java/Gumball/Gumball.java:
```diff
@@ -0,0 +1,61 @@
+        ui.intro();
+        while (true) {
+            input = ui.readCommand();
+            if (input.equals("bye")) break;
```
Should the loop body be wrapped by curly brackets?
Should the conditional be put on a separate line?

sarahteonin commented on src/main/java/Gumball/Parser.java:
```diff
@@ -0,0 +1,32 @@
+
+    /**
+     * Returns a Command which can be executed at a later stage.
+     * @param input The user input that determines what command is returned.
```
Should there be a blank line between description and params?

sarahteonin commented on src/main/java/Gumball/TaskList.java:
```diff
@@ -0,0 +1,147 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    public ArrayList<Task> tasks;
```
I like how you remembered to make the array list name plural

sarahteonin reviewed (commented):
Just some minor coding standards you forgot to follow, else good job!

zlch committed:
Update README.md

zlch committed:
Update README.md

zlch committed:
Update README.md

zlch committed:
Update README.md

zlch committed:
Update README.md

zlch committed:
Update README.md

mongj commented on src/main/java/Gumball/AddCommand.java:
```diff
@@ -0,0 +1,51 @@
+    private String input;
+
+    /**
+     *
```
Remeber to add short summary of the method as the first sentence in the javadoc comment. For the other comments as well

mongj commented on src/main/java/Gumball/UI.java:
```diff
@@ -0,0 +1,37 @@
+
+import java.util.Scanner;
+
+public class UI {
```
Class name should be in PascalCase

```suggestion
public class Ui {
```

mongj commented on src/main/java/Gumball/UI.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    public static void print(String out) {
+        String str = "____________________________________________________________";
```
Perhaps a more descriptive name like `divider` could be better?

mongj commented on src/test/java/Gumball/TaskListTest.java:
```diff
@@ -0,0 +1,84 @@
+public class TaskListTest {
+
+    @Test
+    public void add_simpleToDo_Success() throws InputErrorException {
```
Test method names should follow the convention `unitBeingTested_descriptionOfTestInputs_expectedOutcome`, where of the 3 sections separated by underscore are written in camelCase. 

```suggestion
    public void add_simpleToDo_success() throws InputErrorException {
```
Same for the other methods in this file.

mongj commented on src/main/java/Gumball/AddCommand.java:
```diff
@@ -0,0 +1,51 @@
+     * @param list The taskList where the task will be stored.
+     * @param ui Class which contains ui functions.
+     * @param fileManager The location where the information on the list is stored.
+     * @throws InputErrorException
```
Perhaps can include more details on the circumstance under which this exception will be thrown e.g.
```suggestion
     * @throws InputErrorException when attempting to add an invalid <code>Task</code> object to the <code>TaskList</code>
```
Same for the `IOException` below and other comments

mongj commented on src/main/java/Gumball/DeleteCommand.java:
```diff
@@ -0,0 +1,32 @@
+    @Override
+    public void execute(TaskList list, UI ui, FileManager fileManager)
+            throws InputErrorException, IOException {
+        String temp = list.delete(index);
```
Could use a more descriptive name instead of `temp`, perhaps `deletedTask`?

mongj commented on src/main/java/Gumball/Event.java:
```diff
@@ -0,0 +1,25 @@
+import java.time.LocalDate;
+
+public class Event extends Task {
+    private LocalDate eventStart;
```
`eventStart` could be renamed to `start` or `startDate` etc. to be cleaner, since it is assumed that the start attribute of an `Event` class would refer to the start date of that event

mongj commented on src/main/java/Gumball/Event.java:
```diff
@@ -0,0 +1,25 @@
+    public Event(String desc) throws TaskException {
+        super("", desc);
+        try {
+            String[] section = desc.substring(6).split("/from | /to ");
```
Array variable should have a plural name

```suggestion
            String[] sections = desc.substring(6).split("/from | /to ");
```

mongj commented on src/main/java/Gumball/Parser.java:
```diff
@@ -0,0 +1,32 @@
+        } else if (input.startsWith("delete ")) {
+            int num = Integer.parseInt(input.replaceAll("[^0-9]", ""));
+            return new DeleteCommand(num);
+        } else if (input.startsWith("todo")
```
Is there a missing curly brace here?

mongj commented on src/main/java/Gumball/TaskList.java:
```diff
@@ -0,0 +1,147 @@
+     * Returns number of tasks in list.
+     * @return number of tasks in list.
+     */
+    public int getN() {
```
Perhaps could consider using a more descriptive name like `getSize` as it is not immediately obvious to the caller what N here might refer to (and neither is it mentioned in the comment)

mongj reviewed (commented):
LGTM! Just some nits with regards to naming and more descriptive comments. Otherwise it's good!

zlch committed:
Add Gui features

zlch committed:
Finish Gui features

zlch committed:
Merge branch 'master' of https://github.com/zlch/ip

zlch committed:
Add additional javadoc
Exit command lacked javadoc comment added to increase readability

zlch committed:
Add more javadoc comments

There are many different commands.
In order for users to know what each command does
more specific information should be given.

zlch committed:
Improve code quality in Deadline and AddCommand, improve exception handling in Deadline

Improved the code quality in the AddCommand class by removing repeat statements
and transferring the updateFile function to the addToList class.
Improved code quality in Deadline by moving the formatting of the input String into
another function to increase readabilty through abstraction

zlch committed:
Improve code quality in DeleteCommand and Event

String is being formatted withing the primary function

Code is harder to read

Let's put the String formatted into a private function instead

zlch committed:
Improve code quality

Name of functions in tasklist are not informative.

More descriptive function names make it easier to understand the code.

Let's change the names to be more descriptive.

UI is an argument for multiple functions despite not being used.

This makes the code less readable and may confuse users expecting UI to be used.

Let's remove the UI from the function arguments.

zlch committed:
Increase level of abstraction in AddCommand and FileManager

There was code of different levels of abstraction in Addcommand and FileManager.

Which makes the code harder to read and understand.

Let's create more private methods to increase level of abstraction for the areas with lower level abstraction.

zlch committed:
Add Assertions

No Assertions for Tasks.

Assertions make it easier to ensure code is running properly.

Let's add Assertions to ensure the input is correct.

zlch committed:
Merge pull request #2 from zlch/branch-A-Assertions

Add Assertions

zlch committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	gumball.txt
#	src/main/java/Gumball/Deadline.java
#	src/main/java/Gumball/Event.java

zlch committed:
Merge pull request #3 from zlch/branch-A-CodeQuality

Branch a code quality

zlch committed:
Add additional task FixedDurationTask

No task type for tasks with no specific timing but a duration.

Unable to properly represent these types of tasks.

Let's add another type of tasks that takes in a description and a duration.

zlch committed:
Merge branch 'B-FixedDurationTasks'

zlch committed:
Add exception handling for the Mark and Delete Commands.

Lack of exception handling for when user inputted string instead of integer.

This lead to users not being able to tell the reason why the command was not being processed.

Let's change input of MarkCommand and DeleteCommand to Strings and check the input within the individual classes.

zlch committed:
Improve code in Parser

zlch committed:
Improve code in MarkCommand

zlch committed:
Improve code in Gumball

zlch committed:
Improve code in FixedDurationTask

zlch committed:
Improve code in Deadline
